File: IntentReceiver/src/main/java/com/lilong/intentreceiver/activity/SecondActivity.java
package com.lilong.intentreceiver.activity;

import android.app.Activity;
import android.os.Bundle;

import com.lilong.intentreceiver.R;

public class SecondActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);
        setTitle("IntentReceiver : SecondActivity");
    }
}

File: ParcelableTest/src/main/java/com/lilong/parcelabletest/SecondActivity.kt
package com.lilong.parcelabletest

import android.app.Activity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_second.*
import org.jetbrains.anko.onClick

/**
 * 从MainActivity传来的Parcel数据，可在这里复原成Car对象
 * */
class SecondActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)
        setTitle("SecondActivity")
        btnDeParcelIncomingData.onClick {
            intent.extras.classLoader
            var car: Car = intent.getParcelableExtra<Car>("car")
            tvAfterDeParcel.text = car.toString()
        }
    }
}